tests/integration/__init__.py

```diff
@@ -144,23 +144,33 @@ def cleanup_runtime_config_instance(to_cleanup):
 
 atexit.register(cleanup_runtime_config_instance, RUNTIME_CONFIGS)
 
-_SELECTED_PORTS = set()
+_RUNTESTS_PORTS = {}
 
 
 def get_unused_localhost_port():
     '''
     Return a random unused port on localhost
     '''
     usock = socket.socket(family=socket.AF_INET, type=socket.SOCK_STREAM)
+    usock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
     usock.bind(('127.0.0.1', 0))
     port = usock.getsockname()[1]
-    if port in _SELECTED_PORTS:
+    if port in _RUNTESTS_PORTS:
         port = get_unused_localhost_port()
```

Comment:
With this new logic,
```
if port in _RUNTESTS_PORTS:
```
should always be false since the `bind` should always return an unused port. So this code and the recursive call can be removed.

Note that if it could be true, this code below would malfunction:

```
_RUNTESTS_PORTS[port] = usock
```
since you would be binding a new port to an old socket (when the recursive call unwinds). Luckily, it can't be true.